### PR TITLE
Encrypt user modified password

### PR DESCRIPTION
**Pull request by futuretea**
#320

## Problem
No encryption for the modified password

## Solution
Implement the `Update` method to encrypt the modified password

### CODE REVIEW

**Comment by gitlawr** [comment]
~~We used to drop this field on GET. Can you check why it is not working in https://github.com/rancher/harvester/blob/master/pkg/api/user/formatter.go#L13?~~ NVM

File: pkg/api/user/store.go [hunk 1]
```diff
@@ -69,6 +69,18 @@ func (s *userStore) create(request *types.APIRequest, schema *types.APISchema, d
...
+func (s *userStore) Update(request *types.APIRequest, schema *types.APISchema, data types.APIObject, id string) (types.APIObject, error) {
+	newData := data.Data()
+	passwordPlainText := newData.String(FieldPassword)
+	password, err := pkguser.HashPasswordString(passwordPlainText)
+	if err != nil {
+		return types.APIObject{}, apierror.NewAPIError(validation.ServerError, "Failed to encrypt password")
+	}
+	newData.Set(FieldPassword, password)
+	data.Object = newData
+	return s.Store.Update(request, request.Schema, data, id)
```
Comment (gitlawr) [inline pkg/api/user/store.go]: We may want to return the updated object without the password field like L53?

Comment (futuretea) [reply pkg/api/user/store.go]: fixed

**Review by gitlawr — APPROVED** [review]
LGTM

**Review by guangbochen — APPROVED** [review]

**Pull request merged**